packages/components/text/src/Text.tsx

```diff
@@ -119,14 +125,30 @@ export const Text = compressible<TextProps, TextTokens>((props: TextProps, useTo
 
   // return a continuation function that allows this text to be compressed
   return (extra: TextProps, children: React.ReactNode) => {
+    // [TODO(#2268): Remove once RN Core properly supports Dynamic Type scaling
+    interface ScaleStyleAdjustmentProps {
+      fontSize?: number;
+      lineHeight?: number;
+    }
+    let scaleStyleAdjustmentProps: ScaleStyleAdjustmentProps = {};
+    if (dynamicTypeVariant !== undefined && typeof tokenStyle.fontSize === 'number' && typeof tokenStyle.lineHeight === 'number') {
+      const scaleFactor = fontMetricsScaleFactors[dynamicTypeVariant] ?? 1;
+      scaleStyleAdjustmentProps = {
+        fontSize: tokenStyle.fontSize * scaleFactor,
+        lineHeight: tokenStyle.lineHeight * scaleFactor,
+      };
+    }
+    // ]TODO(#2268)
+
     const mergedProps = {
       ...rest,
       ...keyProps,
       ...filteredProps,
       ...extra,
       onPress,
       numberOfLines: truncate || !wrap ? 1 : 0,
-      style: mergeStyles(tokenStyle, props.style, extra?.style),
+      style: mergeStyles(tokenStyle, props.style, extra?.style, scaleStyleAdjustmentProps),
+      allowFontScaling: dynamicTypeVariant === undefined, // TODO(#2268): Remove once RN Core properly supports Dynamic Type scaling
```

Comment:
You could also do:

`....(Platform.OS === 'ios') && {allowFontScaling: true}`

Reply:
Good idea. It also allows us to leave the test snapshot alone.